src/fs/utils.rs

```diff
@@ -277,3 +277,46 @@ pub fn add_motionlist_patch<P: AsRef<Path>, Q: AsRef<Path>>(tree: &mut Tree<ApiL
     error!("Could not add file {} to API tree. Reason: This is not a motion_list.bin file.", full_path.display());
     None
 }
+
+pub fn add_bgm_property_patch<P: AsRef<Path>, Q: AsRef<Path>>(tree: &mut Tree<ApiLoader>, phys_root: P, local: Q) -> Option<Hash40> {
+    let local = local.as_ref();
+    let base_local = local.with_extension("bin");
+
+    let base_local = if let Some(name) = base_local.file_name().and_then(|os_str| os_str.to_str()) {
+        if let Some(idx) = name.find('+') {
+            let mut new_name = name.to_string();
+            new_name.replace_range(idx..idx + 6, "");
+            base_local.with_file_name(new_name)
+        } else {
+            base_local
+        }
+    } else {
+        base_local
+    };
+    let full_path = phys_root.as_ref().join(local); // need the full path so that our API loader can load it
+    if let Some(name) = full_path.file_name() {
+        if name.to_str().unwrap().contains(&"bgm_property") {
+            match base_local.smash_hash() {
+                Ok(hash) => {
+                    tree.insert_file("api:/patch-bgm_property", &base_local);
+                    tree.loader.push_entry(hash, Path::new("api:/patch-bgm_property"), ApiCallback::None);
+                    // We need to add our file to the vector of patch files
+                    tree.loader.insert_bgm_property_patch(hash, &full_path);
+                    if let Some(local) = local.to_str() {
+                        hashes::add(local);
+                    }
+                    if let Some(base_local) = base_local.to_str() {
+                        hashes::add(base_local);
+                    }
+                    return Some(hash);
+                },
+                Err(e) => {
+                    error!("Could not add file {} to API tree. Reason: {:?}", full_path.display(), e);
+                    return None;
+                },
+            }
+        }
+    }
+    error!("Could not add file {} to API tree. Reason: This is not a motion_list.bin file.", full_path.display());
+    None
+}
```

Comment:
This file is intended for small standalone methods that don't rely on anything else, probably keep this in the same file as the patching like the rest or split them in smaller modules if you find it to be too clogged